Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -4,103 +4,132 @@
+        internal const string kInputManagerAssetPath = "ProjectSettings/InputManager.asset";
+        internal const string kAssetPath = "ProjectSettings/InputSystemActions.inputactions";
 
-        static string s_DefaultAssetPath = kDefaultAssetPath;
-        static string s_AssetPath = kAssetPath;
+        static InputActionAsset s_TestAsset = null;
 
 #if UNITY_INCLUDE_TESTS
-        internal static void SetAssetPaths(string defaultAssetPath, string assetPath)
+        internal static void SetTestAsset(InputActionAsset testAsset)
         {
-            s_DefaultAssetPath = defaultAssetPath;
-            s_AssetPath = assetPath;
+            s_TestAsset = testAsset;
         }
 
-        internal static void Reset()
+        internal static void ResetTestAsset()
         {
-            s_DefaultAssetPath = kDefaultAssetPath;
-            s_AssetPath = kAssetPath;
+            s_TestAsset = null;
         }
 
 #endif
 
+        // Ensures the project-wide settings are initialized early.
         [InitializeOnLoadMethod]
         internal static void InstallProjectWideActions()
         {
             GetOrCreate();
         }
 
-        internal static InputActionAsset GetOrCreate()
+        internal static InputActionAsset LoadFromProjectSettings()
         {
-            var objects = AssetDatabase.LoadAllAssetsAtPath(s_AssetPath);
-            if (objects != null)
+            string text;
+            try
+            {
+                text = File.ReadAllText(kAssetPath);
+            }
+            catch
             {
-                var inputActionsAsset = objects.FirstOrDefault(o => o != null && o.name == kAssetName) as InputActionAsset;
-                if (inputActionsAsset != null)
-                    return inputActionsAsset;
+                return null;
             }
 
-            return CreateNewActionAsset();
+            var asset = ScriptableObject.CreateInstance<InputActionAsset>();
+            asset.name = InputSystem.kProjectWideActionsAssetName;
+            try
+            {
+                asset.LoadFromJson(text);
+            }
+            catch (Exception exception)
+            {
+                Debug.LogError($"Could not parse input actions in JSON format from '{kAssetPath}' ({exception})");
+                Object.DestroyImmediate(asset);
+                return null;
+            }
+            return asset;
         }
 
-        private static InputActionAsset CreateNewActionAsset()
+        internal static InputActionAsset LoadFromInputManagerSettings()
         {
-            var json = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, s_DefaultAssetPath));
+            var objects = AssetDatabase.LoadAllAssetsAtPath(kInputManagerAssetPath);
+            if (objects != null)
+                return objects.FirstOrDefault(o => o != null && o.name == InputSystem.kProjectWideActionsAssetName) as InputActionAsset;
```

Comment:
I am not fond of this `FirstOrDefault`.... basically by using this code we are saying that its OK to have multiple assets with name `kProjectWideActionsAssetName` at path `kInputManagerAssetPath`. 
Its fine to select it using approach above with `FirstOrDefault` if we prior to calling that would check that there is only ONE or ZERO such InputActionAsset otherwise I think we should at least write an error message to the console or possibly throw an exception. 
This comment could be ignored if we create a separate PR/ticket on addressing this.